pulp_container/tests/functional/api/test_recursive_add.py

```diff
@@ -1,670 +1,702 @@
-        to_repo_content = self.versions_api.read(
+        to_repo = container_bindings.RepositoriesContainerApi.read(container_repo.pulp_href)
+        to_repo_content = container_bindings.RepositoriesContainerVersionsApi.read(
             to_repo.latest_version_href
         ).content_summary.present
-        self.assertFalse("container.tag" in to_repo_content)
+        assert "container.tag" not in to_repo_content
         # each manifest list is a manifest and references 2 other manifests
-        self.assertEqual(to_repo_content["container.manifest"]["count"], 6)
+        assert to_repo_content["container.manifest"]["count"] == 6
         # each manifest (non-list) has 3 blobs, 1 blob is shared
         # 9th blob is the parent blob from apline repo, which is shared by all other manifests
-        self.assertEqual(to_repo_content["container.blob"]["count"], 9)
+        assert to_repo_content["container.blob"]["count"] == 9
 
-    def test_copy_manifests_by_digest_empty_list(self):
+    @pytest.mark.parallel
+    def test_copy_manifests_by_digest_empty_list(
+        self, container_repo, container_bindings, from_repo
+    ):
         """Passing an empty list copies no manifests."""
-        self.repositories_api.copy_manifests(
-            self.to_repo.pulp_href, {"source_repository": self.from_repo.pulp_href, "digests": []}
+        container_bindings.RepositoriesContainerApi.copy_manifests(
+            container_repo.pulp_href, {"source_repository": from_repo.pulp_href, "digests": []}
         )
-        latest_to = self.repositories_api.read(self.to_repo.pulp_href)
+        latest_to = container_bindings.RepositoriesContainerApi.read(container_repo.pulp_href)
         # Assert a new version was not created
-        self.assertEqual(latest_to.latest_version_href, f"{self.to_repo.pulp_href}versions/0/")
+        assert latest_to.latest_version_href == f"{container_repo.pulp_href}versions/0/"
 
 
-class TestTagCopy(PulpTestCase):
+class TestTagCopy:
     """Test recursive copy of tags content to a repository."""
 
-    @classmethod
-    def setUpClass(cls):
+    @pytest.fixture(scope="class")
+    def from_repo(
+        self,
+        container_repository_factory,
+        container_remote_factory,
+        container_sync,
+        container_bindings,
+    ):
         """Sync pulp/test-fixture-1 so we can copy content from it."""
-        api_client = gen_container_client()
-        cls.repositories_api = RepositoriesContainerApi(api_client)
-        cls.remotes_api = RemotesContainerApi(api_client)
-        cls.versions_api = RepositoriesContainerVersionsApi(api_client)
-        cls.tags_api = ContentTagsApi(api_client)
-        cls.manifests_api = ContentManifestsApi(api_client)
+        remote = container_remote_factory(upstream_name=PULP_FIXTURE_1)
+        repo = container_repository_factory()
+        container_sync(repo, remote)
+        return container_bindings.RepositoriesContainerApi.read(repo.pulp_href)
```

Comment:
If that is exactly the same as above (it looks to me like that), you can define it outside of the classes (still being class scoped.)